x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/AttributeSet.java

```diff
@@ -266,4 +270,25 @@ public void clear() {
             mapBuilder.keySet().clear();
         }
     }
+
+    /**
+     * This class extends {@code Builder}, but its {@code contains} method also matches {@code NamedExpression} instances by their name.
+     * This is useful for Fork plans, where branches may have different Attribute IDs but share a common output schema,
+     * allowing equality checks of used attributes based on their names.
+     */
+    public static class ForkBuilder extends Builder {
```

Comment:
In case you want to avoid IDs, I think you can reuse `Attribute.IdIgnoringWrapper`.

See [here](https://github.com/elastic/elasticsearch/blob/e6a89ed16d550b5f543f4e5fcdce41930d096bbd/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/local/PushExpressionsToFieldLoad.java#L130) for an example